src/accel/SetupOptionsMessenger.cc

```diff
@@ -54,7 +55,15 @@ struct UICommandTraits<T, std::enable_if_t<std::is_integral_v<T>>>
     static std::string to_string(T v) { return std::to_string(v); }
     static long from_string(G4String const& v)
     {
+        // Conversion to long int introduced in Geant4 10.7.0
+#if G4VERSION_NUMBER < 1070
+        G4long vl;
+        std::istringstream is(v);
+        is >> vl;
+        return vl;
+#else
```

Comment:
For compiler preprocessing and regular program flow with if/else, I find it more readable to put the "usual case" first and then the "else" is the exceptional case (in this instance, for backward compatibility).